clang/lib/Sema/SemaBounds.cpp

```diff
@@ -1172,6 +1270,43 @@ namespace {
         return ProofResult::Maybe;
       }
 
+      // InRangeWithFreeVars is the same as InRange, but gathers free variables
```

Comment:
Why do both `InRangeWithFreeVars` and `InRange` exist? Is there a context in which `InRange` would be called but `InRangeWithFreeVars` would not?

Reply:
`InRange` is also called in `ProveMemoryAccessInRange` (https://github.com/microsoft/checkedc-clang/blob/master/clang/lib/Sema/SemaBounds.cpp#L1728), where we don't check free variables.

Reply:
@kkjeer, could you update this comment?